server/src/main/java/org/elasticsearch/action/bulk/BulkPrimaryExecutionContext.java

```diff
@@ -184,6 +184,7 @@ public void resetForExecutionForRetry() {
         currentItemState = ItemProcessingState.INITIAL;
         requestToExecute = null;
         executionResult = null;
+        retryCounter++;
```

Comment:
Coincidentally, this fixes the `retry_on_conflict` parameter of the [update API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html). Currently, it's retrying without any limit when set to a value > 1.

Reply:
Should we put this into a separate PR?

Reply:
No, this PR relies on the retry count so that we retry once on mapping failures. Not doing this leads to a risk of an infinite retry loop.

Reply:
I meant the other PR would go in first as I hope it is non controversial / quick and would have its own changelog entry.

Reply:
Ah, I see. I'd prefer to avoid the overhead of another PR for such a small change but if others also prefer a separate PR, I'll do that.

Reply:
++ on a separate PR, then that specific change can be tested and reviewed separately. Also I suspect it's for data management to review.

Reply:
I've created a separate PR: https://github.com/elastic/elasticsearch/pull/96262